source/work-with-indexes.txt

```diff
@@ -0,0 +1,157 @@
+
+To improve query performance, build indexes on fields that appear often in
+your application's queries and operations that return sorted results. Each
+index that you add consumes disk space and memory when active, so we recommend
+that you track index memory and disk usage for capacity planning. In addition,
+when a write operation updates an indexed field, MongoDB also updates any related
+index.
+
+Because MongoDB supports dynamic schemas, applications can query against fields
+whose names are not known in advance or are arbitrary. MongoDB 4.2 introduced
+:manual:`wildcard indexes </core/index-wildcard/>` to help support these
+queries. Wildcard indexes are not designed to replace workload-based index
+planning.
+
+For more information about designing your data model and choosing indexes appropriate for your application, see the
+:manual:`Data Modeling and Indexes </core/data-model-operations/#indexes>` guide
+in the {+mdb-server+} manual.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_mflix.movies`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see
+:ref:`<cpp-get-started>`.
+
+Create an Index
+---------------
+
+MongoDB supports several different index types to support querying your data.
+The following sections describe the most common index types and provide sample
+code for creating each index type.
+
+Single-Field Indexes
+~~~~~~~~~~~~~~~~~~~~
+
+:manual:`Single-field indexes </core/index-single/>` are indexes with a
+reference to a single field within a collection's
+documents. They improve single-field query and sort performance, and support
+:manual:`TTL (time to live) indexes </core/index-ttl>`, which
+automatically remove documents from a collection after a certain amount of time
+or at a specific clock time. You can specify the sort order of the index entries
+by specifying ``1`` for ascending or ``-1`` for descending.
+
+.. note::
+
+   The ``_id_`` index is an example of a single-field index. This index is
+   automatically created on the ``_id`` field when a new collection is created.
+
+The following example creates a single-field index in ascending order on the ``title`` field:
+
+.. literalinclude:: /includes/indexes/indexes.cpp
+   :start-after: start-index-single
+   :end-before: end-index-single
+   :language: cpp
+   :copyable:
+
+To learn more about single-field indexes, see :manual:`Single Field Indexes
+</core/index-single>` in the {+mdb-server+} manual.
+
+.. _cpp-indexes-remove:
+
+Remove an Index
+---------------
+
+You can remove any unused index except the default unique index on the
+``_id`` field.
+
+The following sections show how to remove a single index and how to remove all
+indexes in a collection.
+
+Remove a Single Index
+~~~~~~~~~~~~~~~~~~~~~
+
+Pass an instance of an index or the index name to the ``drop_index()`` method to
+remove an index from a collection.
```

Comment:
The C++ driver doesn't have a drop_index() method that can be called on a collection. 

Instead, you can obtain an index_view to the collection and call drop_one(), ex. `coll.indexes().drop_one(name)`